source/Handlebars/Compiler/Lexer/Converter/BlockAccumulators/BlockAccumulatorContext.cs

```diff
@@ -77,6 +81,54 @@ private static bool IsPartialBlock (Expression item)
             }
         }
 
+        private static bool IsLooseClosingElement(Expression item, Expression parentItem, out string looseBlockName)
+        {
+            looseBlockName = null;
+
+            var itemBlockName = GetBlockName(item);
+
+            if (itemBlockName == null) return false;
+
+            var parentBlockName = GetBlockName(parentItem);
+
+            if (!itemBlockName.StartsWith("/")) return false;
+
+            if (parentBlockName == null || IsClosingBlockNotMatchParentBlock(itemBlockName, parentBlockName))
+            {
+                looseBlockName = itemBlockName;
+
+                return true;
+            }
+
+            return false;
+        }
+
+        private static bool IsClosingBlockNotMatchParentBlock(string itemBlockName, string parentBlockName)
+        {
+            if (itemBlockName == null) throw new ArgumentNullException(nameof(itemBlockName));
+            if (parentBlockName == null) throw new ArgumentNullException(nameof(parentBlockName));
+
+            if (!parentBlockName.StartsWith("#") || parentBlockName.StartsWith("#>") || parentBlockName.StartsWith("#*")) return false;
+
+            return parentBlockName.Substring(1) != itemBlockName.Substring(1);
+        }
+
+        private static string GetBlockName(Expression item)
+        {
+            item = UnwrapStatement(item);
+            switch( item )
+            {
+                case PathExpression pathExpression:
+                    return pathExpression.Path;
+
+                case HelperExpression helperExpression:
+                    return helperExpression.HelperName;
+
+                default:
+                    return null;
+            }
```

Comment:
nit: Use [switch expression](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression#:~:text=switch%20expression%20%28C%23%20reference%29%201%20Case%20guards.%20A,an%20exception.%20...%203%20C%23%20language%20specification.%20)

Reply:
Yeah, as I did it initially, but then I noticed another code block with this style and formated my code similarly because of
> Contributing
> ...
> Follow the established code format

Reply:
I'd prefer to utilize new features when applicable 😄

Reply:
Me too! Done.